app/data/ContributionData.scala

```diff
@@ -0,0 +1,114 @@
+package data
+
+import anorm._
+import data.AnormMappings._
+import models.{ContributionMetaData, Contributor, PaymentHook}
+import play.api.db.Database
+
+class ContributionData(db: Database) {
+  def insertPaymentHook(paymentHook: PaymentHook): Unit = {
+    db.withConnection(autocommit = true) { implicit conn =>
+      val request = SQL"""
+        INSERT INTO payment_hooks(
+          contributionid,
+          paymentid,
+          provider,
+          created,
+          currency,
+          cardcountry,
+          amount,
+          convertedamount,
+          status,
+          email
+        ) VALUES (
+          ${paymentHook.contributionId}::uuid,
+          ${paymentHook.paymentId},
+          ${paymentHook.provider}::paymentProvider,
+          ${paymentHook.created},
+          ${paymentHook.currency},
+          ${paymentHook.cardCountry},
+          ${paymentHook.amount},
+          ${paymentHook.convertedAmount},
+          ${paymentHook.status}::paymentStatus,
+          ${paymentHook.email}
+        ) ON CONFLICT(contributionId) DO
+        UPDATE SET
+          contributionid = excluded.contributionid,
+          paymentid = excluded.paymentid,
+          provider = excluded.provider,
+          created = excluded.created,
+          currency = excluded.currency,
+          cardcountry = excluded.cardcountry,
+          amount = excluded.amount,
+          convertedamount = excluded.convertedamount,
+          status = excluded.status,
+          email = excluded.email"""
+      request.execute()
+    }
+  }
+
+  def insertPaymentMetaData(pmd: ContributionMetaData): Unit = {
+    db.withConnection(autocommit = true) { implicit conn =>
+      val request = SQL"""
+        INSERT INTO contribution_metadata(
+          contributionid,
+          created,
+          email,
+          ophanid,
+          abtests,
+          cmp,
+          intcmp
+        ) VALUES (
+          ${pmd.contributionId}::uuid,
+          ${pmd.created},
+          ${pmd.email},
+          ${pmd.ophanId},
+          ${pmd.abTests},
+          ${pmd.cmp},
+          ${pmd.intCmp}
+        ) ON CONFLICT(contributionId) DO
+        UPDATE SET
+          contributionid = excluded.contributionid,
+          created = excluded.created,
+          email = excluded.email,
+          ophanid = excluded.ophanid,
+          abtests = excluded.abtests,
+          cmp = excluded.cmp,
+          intcmp = excluded.intcmp"""
+      request.execute()
+    }
+  }
+
+  def insertContributor(contributor: Contributor): Unit = {
+    db.withConnection(autocommit = true) { implicit conn =>
```

Comment:
I understand the API argument, in fact I started using api gateway / lambdas but stopped midway as it was getting fiddly and painful to use. I took a more pragmatic approach as we're running out of time and double checked with @markjamesbutler

It is not excluded to move that code into an API behind a different domain name, I too like separation of responsibility. As a side note It wouldn't bring more in terms of security as the API would be exposed on the Internet anyway (web hooks) but I would be more elegant to put backend considerations away from the frontend. 

I'm not sure about the queue argument though. I don't think that would answer any of our needs.